code/modules/power/prototype_shield_emitter.dm

```diff
@@ -0,0 +1,199 @@
+/obj/machinery/power/proto_sh_emitter
+	name = "Prototype Shield Emitter"
+	desc = "This is a Prototype Shield Emitter that create in front of it a box made of shielding elements to protect the station from heat and pressure"
+	icon = 'icons/obj/power.dmi'
+	icon_state = "proto_sh_emitter"
+	anchored = FALSE
+	density = TRUE
+	max_integrity = 350
+	integrity_failure = 0.2
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 50
+	circuit = /obj/item/circuitboard/machine/proto_sh_emitter
+	///Store the powered shields placed in the world, used when turned off to removed them
+	var/list/signs
+	///Check if the machine is turned on or off
+	var/is_on = FALSE
+	///Check if the machine is locked
+	var/locked = FALSE
+	///Used to check if the machine is placed inside the borders of the map
+	var/borders = TRUE
+	///Check if the machine is the normal version or the small version
+	var/normal = TRUE
+	///Used to check the area power
+	var/apc_power = TRUE
```

Comment:
Why not turn all these booleans into flags?

Reply:
how do i do that?

Reply:
You define the flags:
```cpp
#define PSE_ON (1<<0)
#define PSE_LOCKED (1<<1)
#define PSE_BORDERS (1<<2)
...
```
Then you remove these boolean flags and substitute it with something like this:
```cpp
var/pse_flags = PSE_BORDERS | PSE_NORMAL | PSE_APCPOWER
```
Then instead of doing checks such as `if(normal)` you do `if(pse_flags & PSE_NORMAL)`
And instead of assigning the value like `normal = TRUE` or `normal = FALSE` you do `pse_flags |= PSE_NORMAL` and `pse_flags &= ^PSE_NORMAL`